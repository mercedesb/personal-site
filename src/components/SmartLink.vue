<template>
  <div>
    <router-link :to="to" v-if="!isExternal">
      <span>
        <slot></slot>
      </span>
    </router-link>
    <a :href="to" target='_blank' v-if="isExternal">
      <slot></slot>
    </a>
  </div>
</template>

<script>
export default {
  props: {
    to: [String, Object],
    isExternal: Boolean
  }
}
</script>

<style lang="scss">
</style>