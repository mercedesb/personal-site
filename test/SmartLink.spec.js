import { shallow } from '@vue/test-utils'
import SmartLink from '@/components/SmartLink.vue'

describe('SmartLink', () => {
  describe('with an internal link', () => {
    it('matches snapshot', () => {
      const to = 'test'
      const wrapper = shallow(SmartLink, {
        propsData: { to }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('with an external link', () => {
    it('matches snapshot', () => {
      const to = 'http://google.com'
      const isExternal = true
      const wrapper = shallow(SmartLink, {
        propsData: { to, isExternal }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})