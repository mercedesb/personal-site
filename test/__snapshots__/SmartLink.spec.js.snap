// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartLink with an external link matches snapshot 1`] = `
<div>
  <!---->
  <a href="http://google.com" target="_blank"></a>
</div>
`;

exports[`SmartLink with an internal link matches snapshot 1`] = `
<div>
  <router-link to="test"><span></span></router-link>
  <!---->
</div>
`;
